huaweicloud/services/fgs/resource_huaweicloud_fgs_function.go

```diff
@@ -1777,19 +1764,67 @@ func updateFunctionVersions(client *golangsdk.ServiceClient, d *schema.ResourceD
 	var (
 		functionUrn = parseFunctionUrnWithoutVersion(d.Id())
 
-		oldSet, newSet = d.GetChange("versions")
-		decrease       = oldSet.(*schema.Set).Difference(newSet.(*schema.Set))
-		increase       = newSet.(*schema.Set).Difference(oldSet.(*schema.Set))
+		oldVal, newVal = d.GetChange("versions")
+		oldVersions    = oldVal.(*schema.Set).List()
+		newVersions    = newVal.(*schema.Set).List()
+
+		err error
 	)
 
-	err := deleteFunctionVersions(client, functionUrn, decrease.List())
-	if err != nil {
-		return fmt.Errorf("error deleting function versions: %s", err)
+	// Version + Alias -> Version: Remove alias
+	// Version + Alias -> null:    Remove version
+	// Version -> null:            Remove version
+	newVersionNames := utils.PathSearch("[*].name", newVersions, make([]interface{}, 0)).([]interface{})
+	for _, oldVersion := range oldVersions {
+		versionName := utils.PathSearch("name", oldVersion, nil)
+		versionDesc := utils.PathSearch("description", oldVersion, nil)
+		if fmt.Sprint(versionName) != "latest" && (!utils.SliceContains(newVersionNames, versionName) ||
+			fmt.Sprint(versionDesc) != utils.PathSearch(fmt.Sprintf("[?name=='%s']|[0].description", versionName), newVersions, "").(string)) {
```

Comment:
This line of conditional judgment is difficult to understand.
 It is recommended to simplify it and add comments to facilitate understanding.

Reply:
supplement more comments